docs/monitors/haproxy.md

```diff
@@ -0,0 +1,207 @@
+<!--- GENERATED BY gomplate from scripts/docs/monitor-page.md.tmpl --->
+
+# haproxy
+
+Monitor Type: `haproxy` ([Source](https://github.com/signalfx/signalfx-agent/tree/master/internal/monitors/haproxy))
+
+**Accepts Endpoints**: **Yes**
+
+**Multiple Instances Allowed**: Yes
+
+## Overview
+
+This monitor scrapes [HAProxy](http://www.haproxy.org/) statistics from an HTTP endpoint or a UNIX socket.
+It requires HAProxy 1.8+ and supports running HAProxy in multi-process mode where the HAProxy process each
+metric pertains to is indicated by the metric dimension `process_num`.
+
+<!--- SETUP --->
+### HTTP Endpoint Config
+HAProxy stats must be enabled as described [here](https://www.haproxy.com/blog/exploring-the-haproxy-stats-page/).
+Define the HAProxy monitor in the agent configuration file and provide the `csv export` URL where HAProxy
+stats are served in CSV format. This monitor supports basic HTTP authentication. Simply provide the
+username and password if required as shown below:
+```
+...
+monitors:
+  - type: haproxy
+    url: "http://localhost:8404/stats?stats;csv"
+    username: "your username here"
+    password: "your password here"
+...
+```
+Note: The HAProxy process level stats given by the [show info](https://cbonte.github.io/haproxy-dconv/1.8/management.html#9.3-show%20info) command
+are not available at the HTTP endpoint. Scrape stats from the UNIX socket instead to include the `show info`
+command metrics.
+
+### UNIX Socket Config
+The location of the HAProxy socket file is defined in the HAProxy config file, as in the following example:
+```
+global
+    daemon
+    stats socket /var/run/haproxy.sock
+    stats timeout 2m
+```
+For the given location of the socket file, configure the HAProxy monitor URL as shown below:
+```
+...
+monitors:
+  - type: haproxy
+    url: "unix:/var/run/haproxy.sock"
+...
+```
+Note: The agent process needs read/write permissions to the HAProxy socket file.
+
+
+## Configuration
+
+To activate this monitor in the Smart Agent, add the following to your
+agent config:
+
+```
+monitors:  # All monitor config goes under this key
+ - type: haproxy
+   ...  # Additional config
+```
+
+**For a list of monitor options that are common to all monitors, see [Common
+Configuration](../monitor-config.md#common-configuration).**
+
+
+| Config option | Required | Type | Description |
+| --- | --- | --- | --- |
+| `username` | no | `string` | Basic Auth username to use on each request, if any. |
+| `password` | no | `string` | Basic Auth password to use on each request, if any. |
+| `url` | **yes** | `string` | URL on which to scrape HAProxy. |
+| `sslVerify` | no | `bool` | Flag that enables SSL certificate verification for the scrape URI. (**default:** `true`) |
+| `timeout` | no | `int64` | Timeout for trying to get stats from HAProxy. This should be a duration string that is accepted by https://golang.org/pkg/time/#ParseDuration (**default:** `5s`) |
```

Comment:
@jchengsfx if you have some time, can you try and fix the rendering of the Duration config fields so they show as `duration` in the docs instead of `int64`?